pkg/clusterstack/factory.go

```diff
@@ -103,11 +113,21 @@ func (f *Factory) UpdateStack(keychain authn.Keychain, stack *v1alpha1.ClusterSt
 }
 
 func (f *Factory) RelocatedBuildImage(keychain authn.Keychain, kpConfig config.KpConfig, tag string) (string, error) {
-	return f.Uploader.UploadedBuildImageRef(keychain, tag, kpConfig.CanonicalRepository)
+	canonicalRepo, err := kpConfig.CanonicalRepository()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get canonical repository")
```

Comment:
Do we need to wrap the error here? We don't appear to elsewhere in the file.

Reply:
i only wrapped it because the test that already existed was expecting it, but im happy to change it

Reply:
Might be a good simplification to avoid wrapping the error multiple times if instead we can provide meaningful error messages when we first create the error.